plugin/openassetio_manager_bal/BasicAssetLibraryInterface.py

```diff
@@ -212,6 +232,10 @@ def isEntityReferenceString(self, someString, hostSession):
 
     @simulated_delay
     def entityExists(self, entityRefs, context, _hostSession, successCallback, errorCallback):
+        if not self.hasCapability(self.Capability.kExistenceQueries):
+            super().entityExists(entityRefs, context, _hostSession, successCallback, errorCallback)
```

Comment:
[nit] I had a little dig around and it looks like `super.entityExists` just throws a not implemented error, which makes sense as this path is the capability not supported path.  To me it feels a little odd that the not supported exception isn't thrown directly here.  The code gives the impression that if the plugin doesn't support a capability, then a default implementation from the super class is used.  If we never plan on adding implementations to ManagerInterface, then it seems weird to call out to the super class just to throw the exception.

Reply:
I see what you mean. Some methods do have implementations in the base class, but most throw a NotImplementedException.  

There is a good argument that calling the base class allows us to change behaviour in future, albeit unlikely.

But the main argument against is effort - we'd have to reproduce the logic for constructing the exception message, or expose that functionality somehow.

Reply:
Makes sense, I'm glad you've thought about it